File: weather.js

var unirest = require("unirest");

//var req = unirest("GET", "https://community-open-weather-map.p.rapidapi.com/weather");
var req = unirest.get('https://community-open-weather-map.p.rapidapi.com/weather').type('json');

req.query({
	//"callback": "test",
	"id": "2172797",
	"units": "\"imperial\"",
	"mode": "xml, html",
	"q": "Beijing, CN",
});

req.headers({
	"x-rapidapi-host": "community-open-weather-map.p.rapidapi.com",
	"x-rapidapi-key": "23876321f5msha2807cb302e400fp108e5djsn4bcd7ff8177f"
});



req.end(function (res) {
	if (res.error) throw new Error(res.error);
			var data = res.body;
			console.log(data['main'].temp_min);
			console.log(data['main'].temp_max);
});
function load(){
  var url="https://community-open-weather-map.p.rapidapi.com/weather";
  var title="";
  var Request=new XMLHttpRequest();
  Request.open('GET',url, false);
  Request.send();


  if(Request.readyState!=4 ||  Request.status!= 200 || Request.responseText===""){
  window.console.error("Request had an error");
  return;
  }
  var information=JSON.parse(Request.responseText);
  for (var i=0;i<9;i++){
    var title= information.main[0].temp;
    console.log(title);
    // var para=document.createElement("P");
    // para.innerHTML=title;
    // document.body.appendChild(para);

  }
}

File: weather.py

var unirest = require("unirest");

var req = unirest("GET", "https://community-open-weather-map.p.rapidapi.com/weather");

req.query({
	"callback": "test",
	"id": "2172797",
	"units": "\"metric\" or \"imperial\"",
	"mode": "xml, html",
	"q": "London,uk"
});

req.headers({
	"x-rapidapi-host": "community-open-weather-map.p.rapidapi.com",
	"x-rapidapi-key": "23876321f5msha2807cb302e400fp108e5djsn4bcd7ff8177f"
});


req.end(function (res) {
	if (res.error) throw new Error(res.error);

	console.log(res.body);
});
{12 items
"coord":{2 items
"lon":-0.13
"lat":51.51
}
"sys":{4 items
"message":0.0223
"country":"GB"
"sunrise":1398055845
"sunset":1398107249
}
"weather":[1 item
0:{...}4 items
]
"base":"cmc stations"
"main":{5 items
"temp":290.04
"humidity":70
"pressure":1003
"temp_min":287.04
"temp_max":293.15
}
"wind":{3 items
"speed":0.51
"gust":3.6
"deg":93
}
"rain":{1 item
"1h":1.02
}
"clouds":{1 item
"all":56
}
"dt":1398100214
"id":2643743
"name":"London"
"cod":200
}
